src/test/html/html-element-reference-scanner_test.ts

```diff
@@ -0,0 +1,152 @@
+        null as any, {analyzer: new Analyzer({urlLoader: loader})});
+
+    async function getUnderlinedText(sourceRange: SourceRange|undefined) {
+      if (!sourceRange) {
+        return 'No source range produced';
+      }
+      return '\n' + await warningPrinter.getUnderlinedText(sourceRange);
+    }
+
+    setup(() => {
+      scanner = new HtmlCustomElementReferenceScanner();
+    });
+
+    test('finds custom element references', async() => {
+      contents = `<html><body>
+          <div>Foo</div>
+          <x-foo a=5 b="test" c></x-foo>
+          <div>
+            <x-bar></x-bar>
+          </div>
+          <h1>Bar</h1>
+        </body></html>`;
+
+      const document = new HtmlParser().parse(contents, 'test-document.html');
+      let visit = async(visitor: HtmlVisitor) => document.visit([visitor]);
+
+      const features = await scanner.scan(document, visit);
+
+      assert.deepEqual(features.map(f => f.tagName), ['x-foo', 'x-bar']);
+
+      assert.deepEqual(
+          features[0].attributes.map(a => [a.name, a.value]),
+          [['a', '5'], ['b', 'test'], ['c', '']]);
+
+      const sourceRanges = await Promise.all(
+          features.map(async f => await getUnderlinedText(f.sourceRange)));
+
+      assert.deepEqual(sourceRanges, [
+        `
+          <x-foo a=5 b="test" c></x-foo>
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`,
+        `
+            <x-bar></x-bar>
+            ~~~~~~~~~~~~~~~`
+      ]);
+
+      const attrRanges = await Promise.all(features.map(
+          async f => await Promise.all(f.attributes.map(
+              async a => await getUnderlinedText(a.sourceRange)))));
+
+      assert.deepEqual(attrRanges, [
+        [
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                 ~~~`,
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                     ~~~~~~~~`,
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                              ~`
+        ],
+        []
+      ]);
+
+      const attrNameRanges = await Promise.all(features.map(
+          async f => await Promise.all(f.attributes.map(
+              async a => await getUnderlinedText(a.nameSourceRange)))));
+
+      assert.deepEqual(attrNameRanges, [
+        [
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                 ~`,
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                     ~`,
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                              ~`
+        ],
+        []
+      ]);
+
+      const attrValueRanges = await Promise.all(features.map(
+          async f => await Promise.all(f.attributes.map(
+              async a => await getUnderlinedText(a.valueSourceRange)))));
+
+      assert.deepEqual(attrValueRanges, [
+        [
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                   ~`,
+          `
+          <x-foo a=5 b="test" c></x-foo>
+                       ~~~~~~`,
+          `No source range produced`
```

Comment:
Oh! Right, now I recall the relevant detail for why we had trouble with this assertion. I recently changed the code to get attribute value ranges to return undefined for boolean attributes.

So when it was failing before that was because you had tested locally without that change merged in, but travis was seeing it with it merged in.